airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -1422,6 +1422,33 @@ definitions:
             - "$ref": "#/definitions/LegacyToPerPartitionStateMigration"
             - "$ref": "#/definitions/CustomStateMigration"
         default: []
+      file_uploader:
+        title: File Uploader
+        description: (experimental) Describes how to fetch a file
+        type: object
+        required:
+          - type
+          - requester
+          - download_target_extractor
+        properties:
+          type:
+            type: string
+            enum: [ FileUploader ]
+          requester:
+            description: Requester component that describes how to prepare HTTP requests to send to the source API.
+            anyOf:
+              - "$ref": "#/definitions/CustomRequester"
+              - "$ref": "#/definitions/HttpRequester"
+          download_target_extractor:
+            description: Responsible for fetching the url where the file is located. This is applied on each records and not on the HTTP response
+            anyOf:
+              - "$ref": "#/definitions/CustomRecordExtractor"
+              - "$ref": "#/definitions/DpathExtractor"
+          file_extractor:
```

Comment:
I figured some API might return the file as a field in a JSON response for example but for now this would not be needed and we should remove it